packages/validator/src/services/attestation.ts

```diff
@@ -52,16 +55,23 @@ export class AttestationService {
     // A validator should create and broadcast the attestation to the associated attestation subnet when either
     // (a) the validator has received a valid block from the expected block proposer for the assigned slot or
     // (b) one-third of the slot has transpired (SECONDS_PER_SLOT / 3 seconds after the start of slot) -- whichever comes first.
-    await Promise.race([sleep(this.clock.msToSlotFraction(slot, 1 / 3), signal), this.waitForBlockSlot(slot)]);
+    await Promise.race([sleep(this.clock.msToSlot(slot + 1 / 3), signal), this.waitForBlockSlot(slot)]);
+    this.metrics?.attesterStepCallProduceAttestation.observe(this.clock.secFromSlot(slot));
+
+    // Beacon node's endpoint produceAttestationData return data is not dependant on committeeIndex.
+    // Produce a single attestation for all committees, and clone mutate before signing
+    const attestationNoCommittee = await this.produceAttestation(slot);
```

Comment:
there could be errors at each of the step:
+ error producing attestation
+ error signing attestation
+ error publishing attestation

we can track each of the error type, or a general AttestationError metric. It could help us detect missed attestations right at the validator side.

Reply:
Added